Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -410,23 +410,37 @@ def get_indicator_publication(indicator: Dict[str, Any]):
         return publications
 
     @staticmethod
-    def change_attack_pattern_to_stix_attack_pattern(indicator: Dict[str, Any]):
+    def change_attack_pattern_to_stix_attack_pattern(indicator: dict[str, Any]):
         indicator['type'] = f'STIX {indicator["type"]}'
         indicator['fields']['stixkillchainphases'] = indicator['fields'].pop('killchainphases', None)
         indicator['fields']['stixdescription'] = indicator['fields'].pop('description', None)
 
         return indicator
 
     @staticmethod
-    def is_sub_report(report_obj: Dict[str, Any]) -> bool:
+    def is_sub_report(report_obj: dict[str, Any]) -> bool:
         obj_refs = report_obj.get('object_refs', [])
-        for obj_ref in obj_refs:
-            if obj_ref.startswith('report--'):
-                return False
-        return True
+        return all(not obj_ref.startswith("report--") for obj_ref in obj_refs)
 
     @staticmethod
-    def get_ioc_type(indicator: str, id_to_object: Dict[str, Dict[str, Any]]) -> str:
+    def parse_report_relationships(report_obj: dict[str, Any]) -> list[EntityRelationship]:
+        obj_refs = report_obj.get('object_refs', [])
+        relationships: list[EntityRelationship] = []
+        for related_obj in obj_refs:
+            relation_type_and_id = related_obj.split('--')
+            relationships.append(
+                EntityRelationship(
+                    name='related-to',
+                    entity_a=report_obj.get('name'),
+                    entity_a_type='Report',
+                    entity_b=relation_type_and_id[1],
```

Comment:
We can't relate to a hash, its an invalid value for all types. We have to search the actual STIX envelope for the object.